pkg/apis/psmdb/v1alpha1/perconaservermongodbrestore_types.go

```diff
@@ -48,3 +55,19 @@ type PerconaServerMongoDBRestoreList struct {
 	metav1.ListMeta `json:"metadata,omitempty"`
 	Items           []PerconaServerMongoDBRestore `json:"items"`
 }
+
+func (r *PerconaServerMongoDBRestore) CheckFields() error {
+	if len(r.Spec.ClusterName) == 0 {
+		return fmt.Errorf("spec clustereName field is empty")
+	}
+	if len(r.Spec.BackupName) == 0 && len(r.Spec.StorageName) == 0 && len(r.Spec.Destination) == 0 {
+		return fmt.Errorf("fields backupName or StorageNmae and destination is empty")
+	}
+	if len(r.Spec.Destination) > 0 && len(r.Spec.StorageName) == 0 {
+		return fmt.Errorf("spec storageName field is empty")
+	}
+	if len(r.Spec.StorageName) > 0 && len(r.Spec.Destination) == 0 {
+		return fmt.Errorf("spec destination field is empty")
+	}
+	return nil
+}
```

Comment:
What invariants are legal for `BackupName`, `StorageName` and `Destination`? Am I right that either `BackupName` or `StorageName` and `Destination` should be set? If so, current code will fail, for example, with the following config:
```yaml
...
BackupName: bcp-name
Destination: 
StorageName: some-storage
```
Besides, the current code is too long-winded.

Instead of L63-71 should be enough:
```go
if len(r.Spec.BackupName) == 0 && (len(r.Spec.StorageName) == 0 || len(r.Spec.Destination) == 0) {
    return ...
}
```